crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -237,37 +237,38 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
             pre_compiled_deps.is_some(),
             "Must populate 'pre_compiled_deps' with Sui framework"
         );
-        let (additional_mapping, account_names, protocol_config) = match task_opt.map(|t| t.command)
-        {
-            Some((
-                InitCommand { named_addresses },
-                SuiInitArgs {
-                    accounts,
-                    protocol_version,
-                    max_gas,
-                },
-            )) => {
-                let map = verify_and_create_named_address_mapping(named_addresses).unwrap();
-                let accounts = accounts
-                    .map(|v| v.into_iter().collect::<BTreeSet<_>>())
-                    .unwrap_or_default();
-                let mut protocol_config = if let Some(protocol_version) = protocol_version {
-                    ProtocolConfig::get_for_version(protocol_version.into(), Chain::Unknown)
-                } else {
-                    ProtocolConfig::get_for_max_version()
-                };
-                if let Some(mx_tx_gas_override) = max_gas {
-                    protocol_config.set_max_tx_gas_for_testing(mx_tx_gas_override)
+        let (additional_mapping, account_names, mut protocol_config) =
+            match task_opt.map(|t| t.command) {
+                Some((
+                    InitCommand { named_addresses },
+                    SuiInitArgs {
+                        accounts,
+                        protocol_version,
+                        max_gas,
+                    },
+                )) => {
+                    let map = verify_and_create_named_address_mapping(named_addresses).unwrap();
+                    let accounts = accounts
+                        .map(|v| v.into_iter().collect::<BTreeSet<_>>())
+                        .unwrap_or_default();
+                    let mut protocol_config = if let Some(protocol_version) = protocol_version {
+                        ProtocolConfig::get_for_version(protocol_version.into(), Chain::Unknown)
+                    } else {
+                        ProtocolConfig::get_for_max_version()
+                    };
+                    if let Some(mx_tx_gas_override) = max_gas {
+                        protocol_config.set_max_tx_gas_for_testing(mx_tx_gas_override)
+                    }
+                    (map, accounts, protocol_config)
                 }
-                (map, accounts, protocol_config)
-            }
-            None => (
-                BTreeMap::new(),
-                BTreeSet::new(),
-                ProtocolConfig::get_for_max_version(),
-            ),
-        };
+                None => (
+                    BTreeMap::new(),
+                    BTreeSet::new(),
+                    ProtocolConfig::get_for_max_version(),
+                ),
+            };
 
+        protocol_config.set_shared_object_deletion();
```

Comment:
Is this intended to stay like this, or is this a temporary change while you're testing your PR?  Not a blocker for this PR, but we probably want a better story for running transactional tests with/without new features in a more straightforward way.